app/services/import/determine-supplementary-billing-flags.service.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+/**
+ * Determines if an imported licence should be flagged for supplementary billing
+ * @module DetermineSupplementaryBillingFlagsService
+ */
+
+const ProcessImportedLicenceService = require('../licences/supplementary/process-imported-licence.service.js')
+const LicenceModel = require('../../models/licence.model.js')
+
+/**
+ * Determines if an imported licence should be flagged for supplementary billing.
+ *
+ * This service is responsible for determining whether a licence imported from NALD should be flagged for
+ * supplementary billing. It compares the licences end dates (such as lapsed, revoked or expired dates) between WRLS
+ * and the transformed data, and if there is a change in the dates allows the licence to go on to determining the flags.
+ *
+ * @param {object} naldLicence - the legacy NALD licence
+ * @param {object} wrlsLicenceId - the WRLS licence data
+ */
+async function go (naldLicence, wrlsLicenceId) {
+  try {
+    const licenceChanged = await _licenceChanged(naldLicence, wrlsLicenceId)
+
+    if (!licenceChanged) {
+      return
+    }
+
+    await ProcessImportedLicenceService.go(naldLicence, wrlsLicenceId)
+  } catch (error) {
+    global.GlobalNotifier.omfg('Supplementary Billing Flag failed for licence ', wrlsLicenceId, error)
+  }
+}
+
+async function _licenceChanged (naldLicence, wrlsLicenceId) {
+  const query = LicenceModel.query()
+    .select(['id'])
+    .where('id', wrlsLicenceId)
+
+  _whereClauses(query, naldLicence)
+
+  const result = await query
+
+  return result.length === 0
+}
+
+function _whereClauses (query, naldLicence) {
+  const { expiredDate, lapsedDate, revokedDate } = naldLicence
```

Comment:
We need to change the param name (see previous comment)

```suggestion
function _whereClauses (query, importedLicence) {
  const { expiredDate, lapsedDate, revokedDate } = importedLicence
```

I also think it would be useful to add a comment that explains _why_ this method exists, i.e. the problem it is solving.